robosuite/controllers/base_controller.py

```diff
@@ -0,0 +1,98 @@
+import abc
+import numpy as np
+import mujoco_py
+
+
+class Controller(object, metaclass=abc.ABCMeta):
+    """
+    General controller interface.
+
+    Requires reference to mujoco sim object, id_name of specific robot, relevant joint_indexes to that robot, and
+    whether an initial_joint is used for nullspace torques or not
+    """
+    def __init__(self,
+                 sim,
+                 id_name,
+                 joint_indexes,
+                 initial_joint
+    ):
+        self.action_scale = None
+
+        # Note: the following attributes must be specified in extended controller subclasses during the init call:
```

Comment:
It might be useful to make these class properties (method with the @property tag) and make them not implemented in this base class, so that subclasses are forced to implement these.